Index: src/entity/Bomberman.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/entity/Bomberman.java b/src/entity/Bomberman.java
--- a/src/entity/Bomberman.java	
+++ b/src/entity/Bomberman.java	
@@ -17,6 +17,7 @@
     public boolean notMoving;
     public String preDirection;
     public final int ScreenX;
+
     public Bomberman(GamePanel gp, KeyHandler keyH) {
         this.gp = gp;
         this.keyH = keyH;
@@ -35,18 +36,18 @@
 
     public void getPlayerImage() {
         try {
-            up = ImageIO.read(Objects.requireNonNull(getClass().getResourceAsStream("/bomberman/player/player_up.png")));
-            down = ImageIO.read(Objects.requireNonNull(getClass().getResourceAsStream("/bomberman/player/player_down.png")));
-            left = ImageIO.read(Objects.requireNonNull(getClass().getResourceAsStream("/bomberman/player/player_left.png")));
-            right = ImageIO.read(Objects.requireNonNull(getClass().getResourceAsStream("/bomberman/player/player_right.png")));
-            up1 = ImageIO.read(Objects.requireNonNull(getClass().getResourceAsStream("/bomberman/player/player_up_1.png")));
-            up2 = ImageIO.read(Objects.requireNonNull(getClass().getResourceAsStream("/bomberman/player/player_up_2.png")));
-            down1 = ImageIO.read(Objects.requireNonNull(getClass().getResourceAsStream("/bomberman/player/player_down_1.png")));
-            down2 = ImageIO.read(Objects.requireNonNull(getClass().getResourceAsStream("/bomberman/player/player_down_2.png")));
-            left1 = ImageIO.read(Objects.requireNonNull(getClass().getResourceAsStream("/bomberman/player/player_left_1.png")));
-            left2 = ImageIO.read(Objects.requireNonNull(getClass().getResourceAsStream("/bomberman/player/player_left_2.png")));
-            right1 = ImageIO.read(Objects.requireNonNull(getClass().getResourceAsStream("/bomberman/player/player_right_1.png")));
-            right2 = ImageIO.read(Objects.requireNonNull(getClass().getResourceAsStream("/bomberman/player/player_right_2.png")));
+            up = ImageIO.read(Objects.requireNonNull(getClass().getResourceAsStream("/sprites/player/player_up.png")));
+            down = ImageIO.read(Objects.requireNonNull(getClass().getResourceAsStream("/sprites/player/player_down.png")));
+            left = ImageIO.read(Objects.requireNonNull(getClass().getResourceAsStream("/sprites/player/player_left.png")));
+            right = ImageIO.read(Objects.requireNonNull(getClass().getResourceAsStream("/sprites/player/player_right.png")));
+            up1 = ImageIO.read(Objects.requireNonNull(getClass().getResourceAsStream("/sprites/player/player_up_1.png")));
+            up2 = ImageIO.read(Objects.requireNonNull(getClass().getResourceAsStream("/sprites/player/player_up_2.png")));
+            down1 = ImageIO.read(Objects.requireNonNull(getClass().getResourceAsStream("/sprites/player/player_down_1.png")));
+            down2 = ImageIO.read(Objects.requireNonNull(getClass().getResourceAsStream("/sprites/player/player_down_2.png")));
+            left1 = ImageIO.read(Objects.requireNonNull(getClass().getResourceAsStream("/sprites/player/player_left_1.png")));
+            left2 = ImageIO.read(Objects.requireNonNull(getClass().getResourceAsStream("/sprites/player/player_left_2.png")));
+            right1 = ImageIO.read(Objects.requireNonNull(getClass().getResourceAsStream("/sprites/player/player_right_1.png")));
+            right2 = ImageIO.read(Objects.requireNonNull(getClass().getResourceAsStream("/sprites/player/player_right_2.png")));
 
         } catch (IOException e) {
             e.printStackTrace();
