Index: src/entity/Entity.java
===================================================================
diff --git a/src/entity/Entity.java b/src/entity/Entity.java
--- a/src/entity/Entity.java	
+++ b/src/entity/Entity.java	
@@ -1,10 +1,14 @@
 package entity;
 
+import main.GamePanel;
+
 import java.awt.*;
 import java.awt.image.BufferedImage;
 import java.awt.image.BufferedImageFilter;
 
-public abstract class Entity {
+public class Entity {
+
+    GamePanel gp;
     public int x, y;
     public int speed;
 
@@ -16,6 +20,15 @@
     public int spriteCounter = 0;
     public int spriteNum = 1;
 
-    public abstract void update();
-    public abstract void draw(Graphics2D g2);
+    public Entity(GamePanel gp) {
+        this.gp = gp;
+    }
+
+    public void update() {
+
+    }
+
+    public void draw(Graphics2D g2) {
+
+    }
 }
Index: src/entity/Bomberman.java
===================================================================
diff --git a/src/entity/Bomberman.java b/src/entity/Bomberman.java
--- a/src/entity/Bomberman.java	
+++ b/src/entity/Bomberman.java	
@@ -9,10 +9,8 @@
 import java.awt.image.BufferedImage;
 import java.io.IOException;
 import java.util.Objects;
-import java.util.logging.Handler;
 
 public class Bomberman extends Entity {
-    GamePanel gp;
     KeyHandler keyH;
     public static final int intervalImageChange = 9;
     public boolean notMoving;
@@ -32,7 +30,7 @@
     public int frameExplosion = 0, intervalExplosion = 4, indexAniExplosion = 0;
 
     public Bomberman(GamePanel gp, KeyHandler keyH) {
-        this.gp = gp;
+        super(gp);
         this.keyH = keyH;
         preDirection = "down";
         setDefaultValues();
@@ -215,15 +213,15 @@
                 int bombScreenX = bomb.x - x + gp.screenWidth / 2;
                 bombX = bombScreenX;
                 g2.drawImage(bombing[indexAniBomb], bombScreenX, bomb.y, gp.tileSize, gp.tileSize, null);
-            } else if (bomb.x >= gp.screenWidth / 2 && bomb.x <= gp.wordWidth - gp.screenWidth / 2 && x >= gp.screenWidth / 2 && x <= gp.wordWidth - gp.screenWidth / 2) {
+            } else if (bomb.x >= gp.screenWidth / 2 && bomb.x <= gp.worldWidth - gp.screenWidth / 2 && x >= gp.screenWidth / 2 && x <= gp.worldWidth - gp.screenWidth / 2) {
                 int bombScreenX = ScreenX + bomb.x - x;
                 bombX = bombScreenX;
                 g2.drawImage(bombing[indexAniBomb], bombScreenX, bomb.y, gp.tileSize, gp.tileSize, null);
-            } else if (x > gp.wordWidth - gp.screenWidth / 2) {
-                int bombScreenX = bomb.x - gp.wordWidth + gp.screenWidth;
+            } else if (x > gp.worldWidth - gp.screenWidth / 2) {
+                int bombScreenX = bomb.x - gp.worldWidth + gp.screenWidth;
                 bombX = bombScreenX;
                 g2.drawImage(bombing[indexAniBomb], bombScreenX, bomb.y, gp.tileSize, gp.tileSize, null);
-            } else if (bomb.x > gp.wordWidth - gp.screenWidth / 2 && x <= gp.wordWidth - gp.screenWidth / 2) {
+            } else if (bomb.x > gp.worldWidth - gp.screenWidth / 2 && x <= gp.worldWidth - gp.screenWidth / 2) {
                 int bombScreenX = bomb.x - x + gp.screenWidth / 2;
                 bombX = bombScreenX;
                 g2.drawImage(bombing[indexAniBomb], bombScreenX, bomb.y, gp.tileSize, gp.tileSize, null);
@@ -237,13 +235,13 @@
             }
         }
 
-        if (x >= gp.screenWidth / 2 && x <= gp.wordWidth - gp.screenWidth / 2) {
+        if (x >= gp.screenWidth / 2 && x <= gp.worldWidth - gp.screenWidth / 2) {
             ScreenX = gp.screenWidth / 2;
             g2.drawImage(image, ScreenX, y, gp.tileSize, gp.tileSize, null);
         } else if (x < gp.screenWidth / 2) {
             g2.drawImage(image, x, y, gp.tileSize, gp.tileSize, null);
         } else {
-            ScreenX = x - gp.wordWidth + gp.screenWidth;
+            ScreenX = x - gp.worldWidth + gp.screenWidth;
             g2.drawImage(image, ScreenX, y, gp.tileSize, gp.tileSize, null);
         }
     }
Index: src/Tiles/TileManager.java
===================================================================
diff --git a/src/Tiles/TileManager.java b/src/Tiles/TileManager.java
--- a/src/Tiles/TileManager.java	
+++ b/src/Tiles/TileManager.java	
@@ -58,7 +58,7 @@
         for (int i = 0; i < gp.maxWorldRow; i++) {
             for (int j = 0; j < gp.maxWorldCol; j++) {
                 char s = map[i][j];
-                if (gp.bomberman.x >= gp.screenWidth/2 && gp.bomberman.x <= gp.wordWidth - gp.screenWidth/2) {
+                if (gp.bomberman.x >= gp.screenWidth/2 && gp.bomberman.x <= gp.worldWidth - gp.screenWidth/2) {
                     int ScreenX = j * gp.tileSize - gp.bomberman.x + gp.screenWidth/2;
                     if (s == ' ') {
                         g2.drawImage(tile[0].image, ScreenX, i * gp.tileSize, gp.tileSize, gp.tileSize, null);
@@ -76,7 +76,7 @@
                         g2.drawImage(tile[2].image, j * gp.tileSize, i * gp.tileSize, gp.tileSize, gp.tileSize, null);
                     }
                 } else {
-                    int ScreenX = j * gp.tileSize - gp.wordWidth + gp.screenWidth;
+                    int ScreenX = j * gp.tileSize - gp.worldWidth + gp.screenWidth;
                     if (s == ' ') {
                         g2.drawImage(tile[0].image, ScreenX, i * gp.tileSize, gp.tileSize, gp.tileSize, null);
                     } else if (s == '#') {
